supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesCacheHitCopyingActor.scala

```diff
@@ -26,46 +31,64 @@ case class JesCacheHitCopyingActor(override val jobDescriptor: BackendJobDescrip
                                    serviceRegistryActor: ActorRef) extends BackendCacheHitCopyingActor {
 
   override val configurationDescriptor = jesConfiguration.configurationDescriptor
+  private lazy val completionPromise = Promise[BackendJobExecutionResponse]()
+  private lazy val jesAttributes = jesConfiguration.jesAttributes
+  private lazy val metadataJobKey = {
+    val jobDescriptorKey: BackendJobDescriptorKey = jobDescriptor.key
+    MetadataJobKey(jobDescriptorKey.call.fullyQualifiedName, jobDescriptorKey.index, jobDescriptorKey.attempt)
+  }
 
-  override def copyCachedOutputs(cachedJobOutputs: JobOutputs): Future[BackendJobExecutionResponse] = {
+  override def copyCachedOutputs(seqOfSimpletons: Seq[WdlValueSimpleton], callOutputFiles: Map[String,String], returnCode: Option[Int]): Future[BackendJobExecutionResponse] = {
     val workflowPaths = initializationData.workflowPaths
-    val jesWorkflowPaths = workflowPaths.toJesCallPaths(jobDescriptor.key)
-    val callRootPath = jesWorkflowPaths.callRootPath
-    val callRootParent = callRootPath.getParent
-    val callRootRoot = callRootPath.getRoot
-
-    //renaming requires some tech talk
-    def renameCallSpecificFiles: List[String] = {
-      val jesFileNames = List(
-        jesWorkflowPaths.returnCodeFilename,
-        jesWorkflowPaths.stderrFilename,
-        jesWorkflowPaths.stdoutFilename,
-        jesWorkflowPaths.jesLogFilename
-      )
-      jesFileNames
-    }
-
     val gcsFileSystem = workflowPaths.gcsFileSystemWithUserAuth
+    val destinationCallRootPath: Path = workflowPaths.toJesCallPaths(jobDescriptor.key).callRootPath
+    val sourceCallRootPath: Path = gcsFileSystem.getPath(callOutputFiles.get("callRootPath").get) //Ruchi: the key should aways be present and never be empty, from my understanding
 
-    def copyFilesInGcs() = {
-      val stringFilePaths = cachedJobOutputs map { case (lqn, jobOutput) => lqn -> jobOutput.wdlValue.valueString }
 
-      stringFilePaths foreach {
-        case (lqn, filePath) => if (filePath.isGcsUrl)
+    def copyFiles = {
+      def copyCachedCallOutputFiles(sourceString: String): Unit = {
+        val sourcePath = gcsFileSystem.getPath(sourceString)
+        PathCopier.copy(sourceCallRootPath, sourcePath, destinationCallRootPath)
+      }
+      seqOfSimpletons ++ callOutputFiles foreach {
```

Comment:
After some thought--I'm not totally sure how I would make this asynchronous without adding another actor into the mix. But I'll dig a little bit further today, this will probably be a very necessary enhancement...